src/main/java/bio/terra/service/snapshotbuilder/SnapshotAccessRequestModel.java

```diff
@@ -33,18 +26,35 @@ public record SnapshotAccessRequestModel(
   @VisibleForTesting
   static String generateSummaryForCriteria(
       SnapshotBuilderProgramDataListCriteria criteria, SnapshotBuilderSettings settings) {
+    SnapshotBuilderProgramDataListOption listOption =
+        (SnapshotBuilderProgramDataListOption)
+            settings.getProgramDataOptions().stream()
+                .filter(
+                    programDataOption ->
+                        Objects.equals(programDataOption.getId(), criteria.getId())
```

Comment:
Is it possible for `id` to be null? If not, using `equals()` directly is less code.
```suggestion
                        programDataOption.getId().equals(criteria.getId())
```